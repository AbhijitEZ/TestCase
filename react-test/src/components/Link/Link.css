.btn-grp{
    display: block;
    background-color: aqua;
}