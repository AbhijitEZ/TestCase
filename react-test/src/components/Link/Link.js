import React, { Component } from "react";
import "./Link.css";
class Link extends Component {
  state = {
    clicked: false,

    todos: []
  };

  componentDidUpdate(prevProps, prevState) {
    if (prevState.clicked !== this.state.clicked) {
      this.fetchData();
    }
  }
  fetchData = async () => {
    try {
      fetch("https://jsonplaceholder.typicode.com/todos/")
        .then(response => response.json())
        .then(json => {
          console.log(json, "json");
          this.setState({ todos: json.map(todo => todo.title) });
        });
    } catch (error) {
      console.log(error);
    }
  };
  render() {
    return (
      <div className="btn-grp">
        <button onClick={() => this.setState({ clicked: true })}>
          {this.props.children}
        </button>
        {this.state.clicked && <p>Data is fetched</p>}
      </div>
    );
  }
}

export default Link;
