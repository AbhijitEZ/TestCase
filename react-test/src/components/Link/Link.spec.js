import React from "react";
import { shallow } from "enzyme";
import Link from "./Link";

describe("Link Test", () => {
  let wrapper;
  
  beforeEach(() => (wrapper = shallow(<Link>Test</Link>)));

  it("should render correctly", () => expect(wrapper).toMatchSnapshot());

  it("should have expected initial value", () => {
    expect(wrapper.instance().state.clicked).toEqual(false);
  });
  it("button click event", () => {
    wrapper.find("button").simulate("click");
    expect(wrapper.instance().state.clicked).toEqual(true);
  });

  it("Asynchronous", async () => {
    let res = await fetch("https://jsonplaceholder.typicode.com/todos/");
    expect(res.statusText).toBe("OK");
  });

});
