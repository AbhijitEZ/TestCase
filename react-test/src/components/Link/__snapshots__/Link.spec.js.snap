// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link Test should render correctly 1`] = `ShallowWrapper {}`;
